server/models/submissions/getAllSubmissions.js

```diff
@@ -0,0 +1,131 @@
+const { pool } = require('../database')
+
+/**
+ *
+ * @param {Array} filterParams
+ * @return {String}
+ */
+function filterQuery(filterParams) {
+  if (!Array.isArray(filterParams)) {
+    return ''
+  }
+  const index = filterParams.find((param) => param.id === 'type')
+  if (index) {
+    filterParams.push(filterParams.splice(index, 1)[0])
+  }
+  return filterParams
+    .map(({ id, value }) =>
+      value.length
+        ? ` ${id === 'type' ? 'HAVING' : 'AND'} ${id} IN (${value
+            .map((val) => (typeof val === 'string' ? `'${val}'` : val))
+            .join(',')}) `
+        : ''
+    )
+    .join('')
+}
+
+/**
+ *
+ * @param {Array} sortParams
+ * @return {String}
+ */
+function sortQuery(sortParams) {
+  if (!Array.isArray(sortParams)) {
+    return ''
+  }
+  return (
+    ' ORDER BY ' +
+    sortParams
+      .map(({ id, desc }) => `${id} ${desc ? 'DESC' : 'ASC'}`)
+      .join(', ')
+  )
+}
+
+/**
+ * @param {*} param0
+ * @param {String} param0.username
+ * @param {Object} param0.params
+ * @param {Object} param0.query
+ * @return {Promise}
+ */
+
+function getAllSubmissions({ username, params, query }) {
+  return new Promise((resolve, reject) => {
+    const { contestId } = params
+    const {
+      search,
+      sort,
+      page,
+      size,
+      download_csv: downloadCSV,
+      filters,
+    } = query
+
+    let countQuery = `SELECT COUNT(*) AS total FROM 
+    (SELECT id, 'subjective' AS type FROM subjective_submissions WHERE contest_id=? `
+    let mainQuery = `SELECT s.id, s.question_id, q.name, s.type, s.username, s.submission_time, s.score, s.judged FROM 
```

Comment:
I think we can use [multiple statement queries](https://github.com/mysqljs/mysql#multiple-statement-queries) to get count without expensive checks

Reply:
I thought of this, but by default, `mysql` package disallows use of multiple statement queries, as it becomes easily prone to very dangerous SQL injections. Know more about [SQL injections here](https://www.w3schools.com/sql/sql_injection.asp)
So, I feel it is better to let it be done using 2 queries, as there is no other way to do it in a single query (without allowing multiple statement queries). 

Also, since, MySQL queries are executed asynchronously, so it wouldn't slow down our server.

Reply:
Yeah..it is vulnerable to sql injections, but, they are escaped anyways when we use arrays to fill in sql query values.